File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/Entities/Message.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities.Entities
{
    public class Message : Entity
    {
        [Key]
        public Guid MessageId { get; set; }
        public Guid UserConversationId { get; set; }
        public string Content { get; set; } = string.Empty;
        public DateTime Time { get; set; }
        public Guid? ReplyId { get; set; }
        public bool IsEdited { get; set; }

        protected override Message Init()
        {
            return new Message();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/entities/message.js

import Entity from "../entity";

class Message extends Entity {
    constructor() {
        super();
        this.MessageId = null;
        this.UserConversationId = null;
        this.Content = null;
        this.Time = null;
        this.ReplyId = null;
        this.IsEdited = null;
    }

    init() {
        return new Message();
    }
}

export default Message;